### PR TITLE
(fix) O3-1886: Add offline patient registration e2e test

### PR DESCRIPTION
**Pull request by RandilaP**
## Requirements

- [ ] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary
- Add an offline patient registration e2e test

## Screenshots

*None.*

## Related Issue
https://issues.openmrs.org/browse/O3-1886

## Other

*None.*

### CODE REVIEW

**Comment by RandilaP** [comment]
@anjula-sack Done

File: e2e/specs/offlinePatientRegistration.spec.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import { test } from '../core';
+import { expect } from '@playwright/test';
+import { PatientRegistrationFormValues, RegistrationAndEditPage, OfflineActionsPage } from '../pages';
+import { deletePatient, getPatient, waitUntilOfflineCacheStabilizes } from '../commands';
+import dayjs from 'dayjs';
+let patientUuid: string;
```
Comment (jayasanka-sack) [inline e2e/specs/offlinePatientRegistration.spec.ts:6]:
```suggestion
let patientUuid: string;

```

File: e2e/specs/offlinePatientRegistration.spec.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  await patientRegistrationPage.goto();
+  await patientRegistrationPage.fillPatientRegistrationForm(formValues);
+  await page.context().setOffline(false);
+  await offlineActionsPage.goto();
+  await offlineActionsPage.sync();
+  await expect(page).toHaveURL(new RegExp('^[\\w\\d:\\/.-]+\\/patient\\/[\\w\\d-]+\\/chart\\/.*$'));
+  const patientUuid = /patient\/(.+)\/chart/.exec(page.url())?.[1] ?? null;
+  const patient = await getPatient(api, patientUuid);
+  const { person } = patient;
+  const { givenName, middleName, familyName, sex } = formValues;
```
Comment (jayasanka-sack) [inline e2e/specs/offlinePatientRegistration.spec.ts:37]:
```suggestion
  };
  await waitUntilOfflineCacheStabilizes(page);
  await page.context().setOffline(true);
  
  await patientRegistrationPage.goto();
  await patientRegistrationPage.fillPatientRegistrationForm(formValues);
  
  await page.context().setOffline(false);
  await offlineActionsPage.goto();
  await offlineActionsPage.sync();
  
  await expect(page).toHaveURL(new RegExp('^[\\w\\d:\\/.-]+\\/patient\\/[\\w\\d-]+\\/chart\\/.*$'));
  
  const patientUuid = /patient\/(.+)\/chart/.exec(page.url())?.[1] ?? null;
  const patient = await getPatient(api, patientUuid);
  const { person } = patient;
  const { givenName, middleName, familyName, sex } = formValues;
```

File: e2e/specs/offlinePatientRegistration.spec.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  await expect(person.preferredAddress.address1).toBe(formValues.address1);
+  await expect(person.preferredAddress.cityVillage).toBe(formValues.cityVillage);
+  await expect(person.preferredAddress.stateProvince).toBe(formValues.stateProvince);
+  await expect(person.preferredAddress.country).toBe(formValues.country);
+  await expect(person.preferredAddress.countyDistrict).toBe(formValues.countyDistrict);
+  await expect(person.attributes[0].display).toBe(`Telephone Number = ${formValues.phone}`);
+});
+test.afterEach(async ({ api }) => {
+  if (patientUuid) {
+    await deletePatient(api, patientUuid);
```
Comment (jayasanka-sack) [inline e2e/specs/offlinePatientRegistration.spec.ts:51]:
```suggestion
  await expect(person.preferredAddress.countyDistrict).toBe(formValues.countyDistrict);
  await expect(person.attributes[0].display).toBe(`Telephone Number = ${formValues.phone}`);
});

test.afterEach(async ({ api }) => {
  if (patientUuid) {
    await deletePatient(api, patientUuid);
```

**Comment by anjula-sack** [comment]
@RandilaP Shall we close this PR?

**Pull request closed without merging**